backend/test/db-handler.js

```diff
@@ -0,0 +1,43 @@
+// tests/db-handler.js - The setup and teardown methods for the test instance of MongoDB.
+
+const mongoose = require("mongoose");
+const { MongoMemoryServer } = require("mongodb-memory-server");
+
+mongoServer = new MongoMemoryServer();
+
```

Comment:
This file is used across all unit and integration tests.

Reply:
Somewhat useful links:
https://github.com/nodkz/mongodb-memory-server#mongodb-in-memory-server

https://dev.to/paulasantamaria/testing-node-js-mongoose-with-an-in-memory-database-32np

https://dev.to/paulasantamaria/testing-node-js-mongoose-with-an-in-memory-database-32np

https://www.freecodecamp.org/news/end-point-testing/

https://mongoosejs.com/docs/deprecations.html